src/Microsoft.DotNet.Darc/src/DarcLib/AzureDevOpsClient.cs

```diff
@@ -413,43 +561,54 @@ public async Task<IList<Check>> GetPullRequestChecksAsync(string pullRequestUrl)
                         new Check(
                             checkState,
                             status["context"]["name"].ToString(),
-                            $"{pullRequestUrl}/{status["id"]}"));
+                            $"https://dev.azure.com/{accountName}/{projectName}/{statusesPath}/{status["id"]}"));
                 }
             }
 
             return statuses;
         }
 
-        public async Task<string> GetPullRequestBaseBranch(string pullRequestUrl)
-        {
-            HttpResponseMessage response = await this.ExecuteGitCommand(HttpMethod.Get, pullRequestUrl, _logger);
-
-            JObject content = JObject.Parse(await response.Content.ReadAsStringAsync());
-            string baseBranch = content["sourceRefName"].ToString();
-            const string refsHeads = "refs/heads/";
-            if (baseBranch.StartsWith(refsHeads))
+        /// <summary>
+        ///     Execute a command on the remote repository.
+        /// </summary>
+        /// <param name="method">Http method</param>
+        /// <param name="accountName">Azure DevOps account name</param>
+        /// <param name="projectName">Project name</param>
+        /// <param name="requestPath">Path for request</param>
+        /// <param name="logger">Logger</param>
+        /// <param name="body">Optional body if <paramref name="method"/> is Put or Post</param>
+        /// <param name="versionOverride">API version override</param>
+        /// <returns>Http response</returns>
+        private async Task<HttpResponseMessage> ExecuteRemoteGitCommand(
```

Comment:
>HttpResponseMessage [](start = 27, length = 19)

Since this is disposable, and no disposing it can tie up active TCP connections, every single call to this method needs to be in a using scope... I don't think any of them are, and we've seen many a build timeout because of not having any available connections.

Reply:
Yep, got this.  Also fixed up the github version of the client.